### PR TITLE
JSON Schema: HTTP Connection Manager Filter

### PR DESCRIPTION
**Pull request by ccaraman**
Provides schema validation for HTTP Connection Manager Filter.

Next PR will extend coverage for access log and HTTP Filters.

### CODE REVIEW

**Comment by ccaraman** [comment]
@lyft/network-team

File: source/common/json/config_schemas.cc [hunk 1]
```diff
@@ -73,6 +73,82 @@ const std::string Json::Schema::CLIENT_SSL_NETWORK_FILTER_SCHEMA(R"EOF(
...
+          "filter" : {"type" : "object"}
+        },
+        "required" : ["path"],
+        "additionalProperties" : false
+      },
+      "tracing" : {
+        "type" : "object",
+        "properties" : {
+          "operation_name" : {"type" : "string"},
+          "type" : {
```
Comment (RomanDzhabarov) [inline source/common/json/config_schemas.cc]: i think i cleaned that up, there should not be "type" at all under "tracing". Where did you see that?

Comment (ccaraman) [inline source/common/json/config_schemas.cc]: This was from last weeks docs. I'll update this.

Comment (RomanDzhabarov) [inline source/common/json/config_schemas.cc]: yeah, this was very recent change. sg.

Comment (ccaraman) [inline source/common/json/config_schemas.cc]: fixed.

File: source/common/json/config_schemas.cc [hunk 2]
```diff
@@ -72,6 +72,78 @@ const std::string Json::Schema::CLIENT_SSL_NETWORK_FILTER_SCHEMA(R"EOF(
...
+const std::string Json::Schema::HTTP_CONN_NETWORK_FILTER_SCHEMA(R"EOF(
+  {
+    "$schema": "http://json-schema.org/schema#",
+    "definitions" : {
+      "access_log": {
+        "type" : "object",
+        "properties" : {
+          "path" : {"type" : "string"},
+          "format" : {"type" : "string"},
+          "filter" : {"type" : "object"}
```
Comment (mattklein123) [inline source/common/json/config_schemas.cc]: within reason I would like to group the schemas together as much as possible. Can we define the access log filter schema here also?

File: source/common/json/config_schemas.cc [hunk 3]
```diff
@@ -126,12 +198,230 @@ const std::string Json::Schema::REDIS_PROXY_NETWORK_FILTER_SCHEMA(R"EOF(
...
+        },
+        "required" : ["pattern", "name"],
+        "additionalProperties" : false
+      }
+    },
+    "properties" : {
+      "name" : {"type" : "string"},
+      "domains": {
+        "type": "array",
+          "items": {"type" : "string"}
```
Comment (mattklein123) [inline source/common/json/config_schemas.cc]: indent

File: source/common/json/config_schemas.cc (hunk 2 above)
Comment (ccaraman) [inline source/common/json/config_schemas.cc]: I am going to add the access log schemas in another pr so I can update the example configs. I can extend this schema when I do that pr (after the http filters). Sounds good?

Comment (mattklein123) [inline source/common/json/config_schemas.cc]: Yup that's fine. Just fix the other nit, otherwise this looks good.

**Pull request merged**